File: PythonDataAdapters/stimulsoft_data_adapters/events/StiConnectionEventArgs.py
"""
Stimulsoft.Reports.JS
Version: 2025.2.1
Build date: 2025.03.20
License: https://www.stimulsoft.com/en/licensing/reports
"""

from ..classes import StiBaseRequest
from ..classes.StiConnectionInfo import StiConnectionInfo
from ..enums import StiDatabaseType
from .StiEventArgs import StiEventArgs


class StiConnectionEventArgs(StiEventArgs):

    database = StiDatabaseType.NONE
    """[enum] The type of the current database connection."""

    driver: str = None
    """Driver used for connection."""

    info: StiConnectionInfo = None
    """Information about the current connection."""

    link: object = None
    """Database connection identifier."""


### Constructor

    def __init__(self, request: StiBaseRequest, database: StiDatabaseType, driver: str, info: StiConnectionInfo):
        super().__init__(request)

        self.database = database
        self.driver = driver
        self.info = info

File: PHPDataAdapters/events/StiConnectionEventArgs.php
<?php
# Stimulsoft.Reports.JS
# Version: 2025.3.3
# Build date: 2025.07.28
# License: https://www.stimulsoft.com/en/licensing/reports
?>
<?php

namespace Stimulsoft\Events;

use Stimulsoft\Enums\StiDatabaseType;
use Stimulsoft\StiConnectionInfo;

class StiConnectionEventArgs extends StiEventArgs
{
    /** @var StiDatabaseType|string [enum] The type of the current database connection. */
    public $database;

    /** @var string Driver used for connection. */
    public $driver;

    /** @var StiConnectionInfo Information about the current connection. */
    public $info;

    /** @var object Database connection identifier. */
    public $link;


### Constructor

    public function __construct($request, $database, $driver, $info)
    {
        parent::__construct($request);

        $this->database = $database;
        $this->driver = $driver;
        $this->info = $info;
    }
}